File: lib/models/land.dart
import 'crop_entry.dart';
import 'expense_entry.dart';
import 'income_entry.dart';
import 'labor_entry.dart';
import 'upad_entry.dart';

class Land {
  int? id;
  String name;
  double size;
  String location;
  double laborRupees;
  double fertilizerKg;
  double income;
  double expenses;
  double cropProductionKg;
  double animalIncome;
  List<CropEntry> cropEntries;
  List<ExpenseEntry> expenseEntries;
  List<IncomeEntry> incomeEntries;
  List<LaborEntry> laborEntries;
  List<UpadEntry> upadEntries;
  bool detailsLoaded;

  Land({
    this.id,
    required this.name,
    required this.size,
    required this.location,
    this.laborRupees = 0,
    this.fertilizerKg = 0,
    this.income = 0,
    this.expenses = 0,
    this.cropProductionKg = 0,
    this.animalIncome = 0,
    List<CropEntry>? cropEntries,
    List<ExpenseEntry>? expenseEntries,
    List<IncomeEntry>? incomeEntries,
    List<LaborEntry>? laborEntries,
    List<UpadEntry>? upadEntries,
    this.detailsLoaded = false,
  }) : cropEntries = cropEntries ?? [],
       expenseEntries = expenseEntries ?? [],
       incomeEntries = incomeEntries ?? [],
       laborEntries = laborEntries ?? [],
       upadEntries = upadEntries ?? [];
}

File: backend_kishan/app/Models/Land.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * @property int $id
 * @property int $user_id
 * @property bool $is_active
 * @property string $land_name
 * @property string $land_size
 * @property string|null $location
 * @property string $labor_rupees
 * @property string $fertilizer_kg
 * @property string $income_total
 * @property string $expense_total
 * @property string $crop_production_kg
 * @property \Illuminate\Support\Carbon|null $created_at
 * @property \Illuminate\Support\Carbon|null $updated_at
 */
class Land extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'is_active',
        'land_name',
        'land_size',
        'location',
        'labor_rupees',
        'fertilizer_kg',
        'income_total',
        'expense_total',
        'crop_production_kg',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'land_size' => 'decimal:2',
            'labor_rupees' => 'decimal:2',
            'fertilizer_kg' => 'decimal:2',
            'income_total' => 'decimal:2',
            'expense_total' => 'decimal:2',
            'crop_production_kg' => 'decimal:2',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(\App\Models\User::class);
    }

    public function incomeEntries(): HasMany
    {
        return $this->hasMany(\App\Models\IncomeEntry::class);
    }

    public function expenseEntries(): HasMany
    {
        return $this->hasMany(\App\Models\ExpenseEntry::class);
    }

    public function cropEntries(): HasMany
    {
        return $this->hasMany(\App\Models\CropEntry::class);
    }

    public function laborEntries(): HasMany
    {
        return $this->hasMany(\App\Models\LaborEntry::class);
    }

    public function upadEntries(): HasMany
    {
        return $this->hasMany(\App\Models\UpadEntry::class);
    }
}
